src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfigValueSerializer.cs

```diff
@@ -39,6 +53,9 @@ public static EditorConfigValueSerializer<T> Default<T>()
             if (typeof(T) == typeof(string))
                 return (EditorConfigValueSerializer<T>)(object)s_string;
 
+            if (typeof(T) == typeof(bool?))
```

Comment:
>if (typeof(T) == typeof(bool?)) [](http://example.com/codeflow?start=12&length=31)

We should be able to handle enums and nullable enums here, instead of having to specify explicitly for each option, no?

Reply:
Are you saying you want to have a many if statement here?
Like
```
If (typeof(T) == typeof(SomeConfigEnum)
{
     return some_serializer;
}
```
I personally feel it's better to let the Option itself handle this.

Reply:
No, you can check `typeof(T).IsEnum`

Reply:
Yeah, but I am it still need to put many static Enum serializers here.
Each enum type would have a different serializer.

Reply:
I don't see why would that be needed.

Wouldn't this work? 

```
if (typeof(T).IsEnum)
{
   return CreateSerializerForEnum<T>();
}

if (Nullable.GetUnderlyingType(typeof(T))?.IsEnum == true)
{
   return CreateSerializerForNullableEnum<T>();
}
```
You'd need to remove `where T : struct, Enum` constraints, but they are not necessary anyways

Reply:
Actually, TryParse requires T : struct :(.
OK. I guess we can't have a default one without boxing. :(